components/binary_sensor/dfrobot_sen0395.rst

```diff
@@ -0,0 +1,274 @@
+
+    switch:
+      # Switch to turn on/off detection
+      - platform: dfrobot_sen0395
+        name: Mmwave Active
+
+.. _dfrobot_sen0395-component:
+
+Hub Component
+-------------
+
+You only need to have the hub component (``dfrobot_sen0395:`` entry) defined if you want to change the sensor's
+settings or read its state via the UART. A :doc:`/components/binary_sensor/gpio` alone is sufficient if you only want
+to determine presence/occupancy. When you define ``dfrobot_sen0395:`` you'll need to have a ``uart:`` entry in
+your configuration with both the TX and RX pins defined and the baud rate must be set to ``115200``.
+
+Multiple instances of this component may be defined if multiple :doc:`/components/uart` components are available:
+
+.. code-block:: yaml
+
+    dfrobot_sen0395:
+      - id: mmWave_1
+        uart_id: uart_1
+      - id: mmWave_2
+        uart_id: uart_2
+      ...
+
+Configuration variables:
+************************
+
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation. Necessary if you want
+  to define multiple instances of this component.
+- **uart_id** (*Optional*, :ref:`config-id`): Manually specify the ID of the :doc:`/components/uart` if you want
+  to use multiple UART buses.
+
+.. _dfrobot_sen0395-binary_sensor:
+
+Binary Sensor
+-------------
+
+.. _dfrobot_sen0395-via_gpio:
+
+Via GPIO
+********
+
+To determine presence, you can simply use a :doc:`/components/binary_sensor/gpio` and connect a single GPIO pin
+from your processor to the IO2 pin of the mmWave radar. Using a dedicated GPIO pin may offer an advantage since
+it may wake the processor from sleep.
+
+If you don't want/need to change the radar's settings and only need to determine presence/occupancy, the
+``dfrobot_sen0395:`` and ``uart:`` components are not necessary.
+
+.. code-block:: yaml
+
+    binary_sensor:
+      - platform: gpio
+        name: Presence Detected via GPIO
+        device_class: motion
+        pin:
+          number: GPIO15
+          mode: INPUT_PULLDOWN
+
+.. _dfrobot_sen0395-via_uart:
+
+Via UART
+********
+
+Connecting the sensor via the serial connection (UART) allows both changing its settings as well as reading its state.
+Note, however, that the UART peripheral cannot wake the processor; if you plan on sleeping the processor, you'll likely
+still need to use the :ref:`GPIO pin <dfrobot_sen0395-via_gpio>` approach described above (in addition to the UART).
+
+First, setup a :doc:`/components/uart` and :ref:`dfrobot_sen0395-component` and then use its binary sensor platform
+to create individual binary sensors for each presence sensor.
+
+.. code-block:: yaml
+
+    binary_sensor:
+      - platform: dfrobot_sen0395
+        name: Presence Detected via UART
+
+Configuration variables:
+************************
+
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **name** (*Optional*, string): The name for the binary sensor.
```

Comment:
Similar to the above comment about `uart` in examples, don't add `id` and `name` to config variables list as users can see them in the "binary sensor" core config page
```suggestion
```